### DISCUSSION TITLE
Search-Results templates ... specialized per Contentype

### DISCUSSION THREAD

**Discussion by MaxGasta** [OrchardCMS/OrchardCore#6997] [Q&A] [answered]
Hello,

"Search-Result.liquid"  renders the "Summary" templates  based on Results contentype.

I'have a NewsItem custom contentype with its summary template:  "Content-NewsItem.Summary.liquid"

I want a specialized template just for Search-Result page.

**How can i call it?**

I tried   "Search-Result-NewsItem.Summary.liquid"  but the page always calls "Content-NewsItem.Summary.liquid"

**Thanks in advance**

**Comment by Skrypt** [comment]
Good one. I need to check.

**Comment by ns8482e** [comment, marked as answer]
`Search-Result.cshtml` creates Display shape for summary for all content items,
`@await DisplayAsync(await DisplayManager.BuildDisplayAsync(contentItem, null, "Summary"))`

As quick workaround, I believe you can change ”Summary” to specialized shape for your NewsItem content type in `Search-Result.cshtml`.

e.g. following change will call view `Content-NewsItem.NewsItemSummary.liquid`

```cshtml
<ul class="list-group">
        @foreach (var contentItem in Model.ContentItems)
        {
             var displayType = "Summary";
             if (contentItem.ContentType == "NewsItem")
                  displayType = "NewsItemSummary";

            <li class="list-group-item">
                   @await DisplayAsync(await DisplayManager.BuildDisplayAsync(contentItem, null, displayType))
            </li>
        }
    </ul>
```

**Reply by MaxGasta** [reply]
Thanks, it works in this way.
At the end I have a "Content.SearchSummary.liquid"  for generic items and "Content-NewsItem.SearchSummary.liquid" for NewItems.
